library/addressable.lisp

```diff
@@ -26,3 +26,13 @@
 
 (define-instance-addressable (List :elt))
 (define-instance-addressable String)
+
+(coalton-toplevel
+  (declare eq-hash (Addressable :obj => :obj -> UFix))
+  (define (eq-hash obj)
+    "Compute a hash for OBJ based on its `eq?' identity.
```

Comment:
this doesn't really make a difference but SXHASH does it based off of EQ so this will collide more often than not in ordinary circumstances